src/js/datamaps.js

```diff
@@ -430,13 +430,25 @@
             .style("stroke-width", options.lineWidth || 1)
         }
 
-        layer.append("text")
-          .attr("x", x)
-          .attr("y", y)
-          .style("font-size", (options.fontSize || 10) + 'px')
-          .style("font-family", options.fontFamily || "Verdana")
-          .style("fill", options.labelColor || "#000")
-          .text( d.id );
+          if (options.customLabelText && options.customLabelText[d.id]) {
+              layer.append("text")
+                  .attr("x", x)
+                  .attr("y", y)
+                  .style("font-size", (options.fontSize || 10) + 'px')
+                  .style("font-family", options.fontFamily || "Verdana")
+                  .style("fill", options.labelColor || "#000")
+                  .text( options.customLabelText[d.id] );
```

Comment:
`.text` should be able to take a function that can act dynamically, so rather than repeating all of the `.attr` and `.style` calls, the logic for custom labels can be encapsulated in the `.text` function like:

```js
layer.append("text")
  .attr("x", x)
 ...
 .text(function() { 
    if (options.customLabelText && options.customLabelText[d.id]) {
      return options.customLabelText[d.id]
   } else {
    return d.id
  }
 })
```